village/src/App.js

```diff
@@ -11,14 +17,30 @@ class App extends Component {
       smurfs: [],
     };
   }
-  // add any needed code to ensure that the smurfs collection exists on state and it has data coming from the server
-  // Notice what your map function is looping over and returning inside of Smurfs.
-  // You'll need to make sure you have the right properties on state and pass them down to props.
+  
+componentDidMount() {
+  axios
+    .get('http://localhost:3333/smurfs')
+    .then(res => this.setState({ smurfs: res.data }))
+    .catch(err => console.log(err))
+}
+
+addSmurf = (e, smurf) => {
+  e.preventDefault();
+  const {name, age, height} = smurf
+
+  axios
+    .post('http://localhost:3333/smurfs', {name, age, height})
+    .then(res => this.setState({ smurf: res.data }))
+    .catch(err => console.log(err))
+}
+
   render() {
     return (
       <div className="App">
-        <SmurfForm />
-        <Smurfs smurfs={this.state.smurfs} />
+        <Route path='/' component={NavBar} />
+        <Route path='/smurf-form' exact render={() => <SmurfForm addSmurf={this.addSmurf}/>}/>
```

Comment:
We will need history.push so we need to spread in `<Route>`'s props
```suggestion
        <Route path='/smurf-form' exact render={props => < {...props} SmurfForm addSmurf={this.addSmurf}/>}/>
```